### PR TITLE
fix and refactor onlinehud

### PR DESCRIPTION
**Pull request by youbitchoc**

### CODE REVIEW

File: Story/OnlineUIComponents/SpectatorHud.cs [hunk 1]
```diff
@@ -46,23 +46,9 @@ public override void Update()
...
-                    }
-
-
-                }
-
-                if (RainMeadow.isArenaMode(out var _))
-                {
-
-                    if (game.arenaOverlay != null || game.pauseMenu != null || game.manager.upcomingProcess != null)
+                    if (game.pauseMenu != null || game.arenaOverlay != null || camera.hud.map.visible || game.manager.upcomingProcess != null)
```
Comment (6fears7) [inline Story/OnlineUIComponents/SpectatorHud.cs:51]: heads up, camera hud map check will break arena

Comment (youbitchoc) [reply Story/OnlineUIComponents/SpectatorHud.cs]: changed it to `camera.hud?.map?.visible is true`

File: Story/OnlineUIComponents/PlayerSpecificOnlineHud.cs [hunk 2]
```diff
@@ -123,20 +123,19 @@ public override void Update()
...
             }
 
-            if (!clientSettings.inGame) return;
+            this.found = false;
+
             if (camera.room == null || !camera.room.shortCutsReady) return;
-            if (abstractPlayer == null)
+            if (!clientSettings.inGame) return;
+            if (clientSettings.avatars.Count == 0) return;
+            if (clientSettings.avatars[0]?.FindEntity(true) is OnlineCreature oc) // TODO: support multiple avatars
```
Comment (youbitchoc) [inline Story/OnlineUIComponents/PlayerSpecificOnlineHud.cs:131]: I don't like this but
1. negligible performance cost
2. not enough spoons to refactor everything

File: Story/OnlineUIComponents/SpectatorHud.cs [hunk 3]
```diff
@@ -46,23 +46,9 @@ public override void Update()
...
-                    }
-
-
-                }
-
-                if (RainMeadow.isArenaMode(out var _))
-                {
-
-                    if (game.arenaOverlay != null || game.pauseMenu != null || game.manager.upcomingProcess != null)
+                    if (game.pauseMenu != null || game.arenaOverlay != null || camera.hud?.map?.visible is true || game.manager.upcomingProcess != null)
```
Comment (6fears7) [inline Story/OnlineUIComponents/SpectatorHud.cs:51]: Assuming you jumped into arena just to sanity verify this won't create issues, refactor lgtm

**Review by 6fears7 — COMMENTED** [review]
Also much better bookean "mutedPlayers"

File: Story/OnlineUIComponents/SpectatorHud.cs (hunk 3 above)
Comment (youbitchoc) [reply Story/OnlineUIComponents/SpectatorHud.cs:51]: yup, tested spectate to work just fine in arena

**Review by 6fears7 — APPROVED** [review]

**Pull request merged**